chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+```
+
+The `ChainValidateIndex` API serves multiple purposes:
+
+1. Validates the chain index at a specific epoch:
+   - Ensures consistency between indexed data and actual chain state
+   - Reports any errors found during validation
+
+2. Optionally backfills missing data:
+   - Backfills data if the index is missing information for the specified epoch
+   - Backfilling only occurs when the `backfill` parameter is set to `true`
+
+3. Detects "holes" in the index:
+   - If `backfill` is `false` and the index lacks data for the specified epoch, the API returns an error indicating missing data
+
+This API is available for use once the Lotus daemon has started with the `ChainIndexer` enabled. However, calling the API for a single epoch at a time can be cumbersome, especially when backfilling or validating the index over a range of historical epochs, such as during a migration. 
+
+To simplify this process, we're also providing a command-line tool in `lotus-shed`.
+
+### The `lotus-shed chainindex validate-backfill` command-line tool
+
+**Note: This command can only be run when the Lotus daemon is already running with the `ChainIndexer` enabled as it depends on the `ChainValidateIndex` RPC API described above.**
+
+The `lotus-shed chainindex validate-backfill` command is a tool for validating and optionally backfilling the chain index over a range of epochs. It wraps the `ChainValidateIndex` API to efficiently process multiple epochs.
+
+#### Usage:
+```
+lotus-shed chainindex validate-backfill --from <start_epoch> --to <end_epoch> [--backfill] [--log-good]
+```
+
+#### Parameters:
+- `--from` (required): The starting epoch (inclusive) for the validation range. Must be greater than 0.
+- `--to` (required): The ending epoch (inclusive) for the validation range. Must be greater than 0 and less than or equal to `from`.
+- `--backfill` (optional, default: true): Whether to backfill missing index entries during validation. 
+- `--log-good` (optional, default: false): Whether to log details for tipsets that have no detected problems.
+
+The command validates the chain index entries for each epoch in the specified range, checking for missing or inconsistent entries. If `--backfill` is enabled (which it is by default), it will attempt to backfill any missing entries using the `ChainValidateIndex` API.
+
+#### Error conditions:
+- If `from` or `to` are invalid (<=0 or `to` > `from`), an error is returned.
+- If the `ChainValidateIndex` API returns an error for an epoch, indicating an inconsistency between the index and chain state, an error message is logged for that epoch.
+
+#### Logging:
+- **Progress is logged every 2880 epochs (1 day worth of epochs) during the validation process.**
+- If `--log-good` is enabled, details are also logged for each epoch that has no detected problems. This includes:
+  - Null rounds with no messages/events.
+  - Epochs with a valid indexed entry.
```

Comment:
I think it would be better to have this logged in the help/usage message of the tool itself rather than getting snapshotted here.  In this doc we can direct someone to type `lotus-shed chainindex validate-backfill -h` for informative usage directions.